src/native/libs/System.IO.Compression.Native/CMakeLists.txt

```diff
@@ -161,6 +181,7 @@ else ()
     endif ()
 
     if(STATIC_LIBS_ONLY)
+        install_static_library(zlibstatic aotsdk nativeaot)
         install_static_library(System.IO.Compression.Native.Aot aotsdk nativeaot)
         install_static_library(System.IO.Compression.Native.Aot.GuardCF aotsdk nativeaot)
```

Comment:
@jkoritzinsky the nativeaot failure [I linked here](https://github.com/dotnet/runtime/pull/102403#issuecomment-2143020722) seems related to these 3 lines. Should zlibstatic be added to the bottom two lines as well?

Reply:
Here's the problem: zlibstatic's debug lib name is zlibstaticd.lib, so our scripts need to account for that.

https://dev.azure.com/dnceng-public/public/_build/results?buildId=694327&view=logs&j=15d34cf8-8614-5f8a-e364-3c281b6c41dd&t=9ddf4123-b773-5dc1-2b69-29721ec5c0a1&l=1397

Reply:
Alternatively, we can patch (and push upstream) a change to remove this line: https://github.com/zlib-ng/zlib-ng/blob/2e3e5f30a05ee7b113c755ad9a075304e02bc328/CMakeLists.txt#L565C9-L565C29

Reply:
Or try `set_target_properties(zlibstatic PRIVATE DEBUG_POSTFIX "")`, which should also work

Reply:
I can do both, and the last suggestion can unblock me at the moment.

Where should I put the set_target_properties call?

Reply:
BTW it seems the 2nd argument should be PROPERTIES instead of PRIVATE.

Reply:
Put it after `FetchContent_MakeAvailable` as we need the target to be available. Also you're right on the syntax.